x/src/cargo.rs

```diff
@@ -106,6 +108,32 @@ impl Cargo {
         self
     }
 
+    /// Passes extra environment variables to x's target command.
+    pub fn envs<I, K, V>(&mut self, vars: I) -> &mut Self
+    where
+        I: IntoIterator<Item = (K, V)>,
+        K: AsRef<OsStr>,
+        V: AsRef<OsStr>,
+    {
+        for tup in vars {
+            let ins = (tup.0.as_ref().to_os_string(), tup.1.as_ref().to_os_string());
+            self.env_additions.push(ins);
```

Comment:
Instead of duplicating the logic here and below in `env` why not just change this to the following: 
```suggestion
            self.env(key, value)
```
And leave the logic of calling `as_ref` and pushing to the internal vec to a single function.